src/storage/disk_manager.c

```diff
@@ -5312,15 +5312,22 @@ disk_vhdr_get_vol_remarks (const DISK_VOLUME_HEADER * vhdr)
 }
 
 /*
- * disk_vhdr_length_of_varfields () - get length of volume header including variable fields
+ * disk_vhdr_get_vol_header_size () - get the size of volume header including variable fields
  *
- * return    : total length
+ * return    : total size
  * vhdr (in) : volume header
  */
 STATIC_INLINE int
-disk_vhdr_length_of_varfields (const DISK_VOLUME_HEADER * vhdr)
+disk_vhdr_get_vol_header_size (const DISK_VOLUME_HEADER * vhdr)
 {
-  return (vhdr->offset_to_vol_remarks + (int) strlen (disk_vhdr_get_vol_remarks (vhdr)));
+  assert (vhdr != NULL);
+
+  const char *remarks = disk_vhdr_get_vol_remarks (vhdr);
+  const int volume_header_size = (remarks + strlen (remarks) + 1) - (char *) (vhdr);
```

Comment:
주소 연산이라 형변화 (char *)는 필요하지 않을 것 같은데, 워닝 발생하나요?

```suggestion
  const int volume_header_size = (remarks + strlen (remarks) + 1) - (vhdr);
```

Reply:
```
[454/1271] Building CXX object sa/CMakeFiles/cubridsa.dir/__/src/storage/disk_manager.c.o
FAILED: sa/CMakeFiles/cubridsa.dir/__/src/storage/disk_manager.c.o
/opt/rh/devtoolset-8/root/usr/bin/c++  -DCUBRID_EXPORTING -DGCC -DI386 -DLINUX -DMAXPATHLEN=1024 -DSA_MODE -DSYSV -DX86 -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -D_LARGEFILE64_SOURCE -D_REENTRANT -Dcubridsa_EXPORTS -I. -I../include -I../src/api -I../src/base -I../src/broker -I../src/communication -I../src/compat -I../src/connection -I../src/executables -I../src/heaplayers -I../src/sp -I../src/loaddb -I../src/method -I../src/monitor -I../src/object -I../src/optimizer -I../src/parser -I../src/replication -I../src/query -I../src/session -I../src/storage -I../src/thread -I../src/transaction -I../src/xasl -I../cubrid-cci/src/cci -I../cubrid-cci/src/tools -I3rdparty/include -I3rdparty/include/expat -I3rdparty/include/jansson -I3rdparty/Source/lz4/lib -I3rdparty/Source/rapidjson/include -I3rdparty/Source/re2 -I/usr/lib/jvm/java/include -I/usr/lib/jvm/java/include/linux -Wall -Wwrite-strings -Wno-cast-qual -Wredundant-decls -Wextra -Wno-unused -Wno-unused-parameter -Wno-format-security -pthread -g -Wall -fno-inline -ggdb -fno-omit-frame-pointer -std=c++17 -fPIC -MD -MT sa/CMakeFiles/cubridsa.dir/__/src/storage/disk_manager.c.o -MF sa/CMakeFiles/cubridsa.dir/__/src/storage/disk_manager.c.o.d -o sa/CMakeFiles/cubridsa.dir/__/src/storage/disk_manager.c.o -c ../src/storage/disk_manager.c
../src/storage/disk_manager.c: In function ‘int disk_vhdr_get_vol_header_size(const DISK_VOLUME_HEADER*)’:
../src/storage/disk_manager.c:5326:61: error: invalid operands of types ‘const char*’ and ‘const DISK_VOLUME_HEADER*’ {aka ‘const disk_volume_header*’} to binary ‘operator-’
   int volume_header_size = (remarks + strlen (remarks) + 1) - (vhdr);
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~
```

위처럼 워닝이 발생합니다